rust/core-lib/src/event_context.rs

```diff
@@ -415,6 +436,75 @@ impl<'a> EventContext<'a> {
                            ed.get_layers().get_merged(), first, last,
                            ed.is_pristine())
     }
+
+    fn do_request_hover(&mut self, request_id: usize, position: Option<ClientPosition>) {
+        if let Some(position) = self.get_resolved_position(position) {
+            self.with_each_plugin(|p| p.get_hover(self.view_id, request_id, &position))
+        }
+    }
+    
+    // Gives the requested position in UTF-8 offset format to be sent to plugin
+    // If position is `None`, it tries to get the current Caret Position and use
+    // that instead
+    fn get_resolved_position(&mut self, position: Option<ClientPosition>) -> Option<Position> {
+        position.map(|p|
+            self.with_view(|view, text| view.line_col_to_offset(text, p.line, p.column)
+        )).or_else(|| self.view.borrow().get_caret_offset())
+        .map(|offset| self.get_position(offset))
+    }
+
+    fn get_position(&mut self, utf8_offset: usize) -> Position {
+        self.with_rope(|rope| {
+            let line = rope.line_of_offset(utf8_offset);
+            let offset_of_line = rope.offset_of_line(line);
+            let col_utf8 = utf8_offset - offset_of_line;
+            let col_utf16 = rope.convert_metrics::<BaseMetric, Utf16CodeUnitsMetric>(utf8_offset) -
+                rope.convert_metrics::<BaseMetric, Utf16CodeUnitsMetric>(offset_of_line);
+
+            Position {
+                utf8_offset,
+                line,
+                col_utf8,
+                col_utf16,
+            }                           
+        })
+    }
+
+    fn get_hover_in_utf8_offset(&mut self, hover: Hover) -> client::Hover {
+        client::Hover {
+            content: hover.content,
+            range: hover.range.map(|r| self.get_range_in_utf8_offset(r))
+        }
+    }
+
+    fn get_range_in_utf8_offset(&mut self, range: Range) -> client::Range {
+        client::Range {
+            start: self.get_utf8_offset_from_position_enum(range.start),
+            end: self.get_utf8_offset_from_position_enum(range.end),
+        }
+    }
+
+    fn get_utf8_offset_from_position_enum(&mut self, position: PositionEnum) -> usize {
```

Comment:
maybe `offset_from_plugin_position`? I appreciate that this is a different type than above, but at least this fn name communicates what we're doing.